src/crypto.rs

```diff
@@ -5,49 +5,54 @@
 mod crypto_test;
 
 use serde::{Deserialize, Serialize};
-use starknet_crypto::{pedersen_hash, poseidon_hash_many, FieldElement};
+use starknet_types_core::felt::Felt;
+use starknet_types_core::hash::{Pedersen, Poseidon, StarkHash as CoreStarkHash};
 
-use crate::hash::{StarkFelt, StarkHash};
+use crate::hash::StarkHash;
 
 /// An error that can occur during cryptographic operations.
 #[derive(thiserror::Error, Clone, Debug)]
 pub enum CryptoError {
     #[error("Invalid public key {0:?}.")]
     InvalidPublicKey(PublicKey),
     #[error("Invalid message hash {0:?}.")]
-    InvalidMessageHash(StarkFelt),
+    InvalidMessageHash(Felt),
```

Comment:
I think that the default format in the error message will be decimal we probably want to have it as hex

Reply:
Fixed all `CryptoError` messages.